enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/state/CoreState.java

```diff
@@ -238,14 +234,14 @@ public synchronized void start() throws IOException, RaftLogCompactedException,
     @Override
     public synchronized void stop() throws Throwable
     {
-        if ( applier.sync( true ) )
-        {
-            flush();
-        }
+        applier.sync( true );
+        flush();
     }
 
-    public synchronized CoreSnapshot snapshot() throws IOException, RaftLogCompactedException
+    public synchronized CoreSnapshot snapshot() throws IOException, RaftLogCompactedException, InterruptedException
     {
+        applier.sync( false );
+
```

Comment:
It syncs the applier so that everything that is in the process of being applied is actually applied before proceeding, so that you get a consistent snapshot taken at the same logical time (log index) for all the snapshots.